### PR TITLE
[scorecard] reformat scorecard olm tests to ginkgo/gomega

### PR DESCRIPTION
**Pull request by varshaprasad96**
**Description of the change:**
Reformat scorecard olm and basic tests to ginkgo/gomega format

### CODE REVIEW

File: internal/scorecard/alpha/tests/bundle_test.go [hunk 1]
```diff
@@ -18,401 +18,381 @@ import (
...
+			crCount int
+			crList  []unstructured.Unstructured
+		)
+
+		It("Check Bundle CRs", func() {
+			crCount = 1
+
+			bundle, err = apimanifests.GetBundleFromDir(testBundle)
+
+			Expect(bundle).NotTo(BeNil())
```
Comment (estroz) [inline internal/scorecard/alpha/tests/bundle_test.go:75]:
```suggestion
			Expect(err).ToNot(HaveOccurred())
```

File: internal/scorecard/alpha/tests/bundle_test.go [hunk 2]
```diff
@@ -18,401 +18,381 @@ import (
...
 
+			crList, err = GetCRs(bundle)
+			Expect(crCount).To(Equal(len(crList)))
+			Expect(err).To(BeNil())
```
Comment (estroz) [inline internal/scorecard/alpha/tests/bundle_test.go:79]:
```suggestion
			Expect(err).ToNot(HaveOccurred())
```

File: internal/scorecard/alpha/tests/bundle_test.go [hunk 3]
```diff
@@ -18,401 +18,381 @@ import (
...
+	})
 
-func TestBundleCRs(t *testing.T) {
-	cases := []struct {
-		bundlePath string
-		crCount    int
-	}{
-		{testBundle, 1},
-	}
+	Describe("Test Bundle Path", func() {
```
Comment (estroz) [inline internal/scorecard/alpha/tests/bundle_test.go:45]: We should not be testing naked upstream code.

File: internal/scorecard/alpha/tests/bundle_test.go [hunk 4]
```diff
@@ -18,401 +18,381 @@ import (
...
-
-	for _, c := range cases {
-		t.Run(c.bundlePath, func(t *testing.T) {
-
-			bundle, err := apimanifests.GetBundleFromDir(c.bundlePath)
-			if err != nil {
-				t.Fatalf("Error getting bundle: %v", err)
-			}
+		BeforeEach(func() {
+			bundle, _ = apimanifests.GetBundleFromDir(testBundle)
```
Comment (estroz) [inline internal/scorecard/alpha/tests/bundle_test.go:89]: Check this error

File: internal/scorecard/alpha/tests/bundle_test.go [hunk 5]
```diff
@@ -18,401 +18,381 @@ import (
...
+		})
 
-			result := c.function(bundle)
-			if result.State != c.state {
-				t.Errorf("%s result State %v expected", result.Name, c.state)
-				return
-			}
+		It("CheckSpecTest", func() {
```
Comment (estroz) [inline internal/scorecard/alpha/tests/bundle_test.go:92]: Idiomatic usage:
```go
		Context("CheckSpecTest", func() {
		    It("returns a pass state", func() {
		        res = CheckSpecTest(bundle)
		        Expect(res.State).To(Equal(scapiv1alpha2.PassState))
		    })
		})
```

**Comment by varshaprasad96** [comment]
Modified based on reviews

File: internal/scorecard/alpha/tests/basic.go [hunk 6]
```diff
@@ -39,13 +40,17 @@ func CheckSpecTest(bundle *apimanifests.Bundle) scapiv1alpha2.ScorecardTestResul
 		return r
 	}
 
+	return checkSpecHelper(crSet, r)
```
Comment (estroz) [inline internal/scorecard/alpha/tests/basic.go:43]: Nit: helper is implied
```suggestion
	return checkSpec(crSet, r)
```

File: internal/scorecard/alpha/tests/basic.go [hunk 7]
```diff
@@ -39,13 +40,17 @@ func CheckSpecTest(bundle *apimanifests.Bundle) scapiv1alpha2.ScorecardTestResul
...
+}
+
+func checkSpecHelper(crSet []unstructured.Unstructured,
```
Comment (estroz) [inline internal/scorecard/alpha/tests/basic.go:46]:
```suggestion
func checkSpec(crSet []unstructured.Unstructured,
```

File: internal/scorecard/alpha/tests/bundle_test.go [hunk 8]
```diff
@@ -18,403 +18,397 @@ import (
 	"path/filepath"
 	"testing"
 
+	. "github.com/onsi/ginkgo"
+	. "github.com/onsi/gomega"
 	apimanifests "github.com/operator-framework/api/pkg/manifests"
 	operatorsv1alpha1 "github.com/operator-framework/api/pkg/operators/v1alpha1"
+	scapiv1alpha2 "github.com/operator-framework/operator-sdk/pkg/apis/scorecard/v1alpha2"
```
Comment (estroz) [inline internal/scorecard/alpha/tests/bundle_test.go:25]: Nit: move this to its own imports section like it was previously.

File: internal/scorecard/alpha/tests/bundle_test.go [hunk 9]
```diff
@@ -18,403 +18,397 @@ import (
...
-			if err != nil {
-				t.Error(err)
-				return
-			}
-			if len(crList) != c.crCount {
-				t.Errorf("Wanted %d CRs but got: %d", c.crCount, len(crList))
-				return
-			}
+			Expect(crCount).To(Equal(len(crList)))
+			Expect(err).ToNot(HaveOccurred())
```
Comment (estroz) [inline internal/scorecard/alpha/tests/bundle_test.go:62]: Nit: always check `err` before everything else as to not mask errors
```diff
crList, err = GetCRs(bundle)
-Expect(crCount).To(Equal(len(crList)))
Expect(err).ToNot(HaveOccurred())
+Expect(crCount).To(Equal(len(crList)))
```

File: internal/scorecard/alpha/tests/bundle_test.go [hunk 10]
```diff
@@ -18,403 +18,397 @@ import (
...
+				Expect(res.State).To(Equal(scapiv1alpha2.PassState))
+			})
+		})
 	})
 
-	crWithoutDescriptor := unstructured.Unstructured{
-		Object: nil,
-	}
+	Describe("Testing OLM Bundle", func() {
+		It("Should pass when test bundle is at the desired location", func() {
```
Comment (estroz) [inline internal/scorecard/alpha/tests/bundle_test.go:114]:
```suggestion
		It("should pass when test bundle is at the desired location", func() {
```

**Pull request merged**